### ISSUE TITLE
Please enable Folder browsing support.

### ISSUE THREAD

**Issue by ccatlett1984** [Sovol3d/SV06-PLUS#1]
The SV06 has the ability to use folders on the SC card, I'd like the same on SV06 Plus.

**Comment by Sovol3d** [comment]
Hello,
What do you mean? SC card?
Could you please describe it more specific and with some pictures to show it?
Best regards,
Jan
website: sovol3d.com
Follow us to get an exclusive 3D printing information
Sovol Official User Group: https://www.facebook.com/groups/sovol3d <https://www.facebook.com/sovol3d/ >
Facebook Page:https://www.facebook.com/sovol3d <https://twitter.com/Sovol3d >
Youtube: https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog <https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog >
------------------------------------------------------------------
From:ccatlett1984 ***@***.***>
Sent At:2023 May 18 (Thu.) 07:06
To:Sovol3d/SV06-PLUS ***@***.***>
Cc:Subscribed ***@***.***>
Subject:[Sovol3d/SV06-PLUS] Please enable Folder browsing support. (Issue #1)
The SV06 has the ability to use folders on the SC card, I'd like the same on SV06 Plus.

**Comment by ccatlett1984** [comment]
Sorry for the typo, I mean the SD card.

**Comment by Sovol3d** [comment]
Hello, 
It can. You can connect it with a USB or card reader. 
Best regards,
Jan
website: sovol3d.com
Follow us to get an exclusive 3D printing information
Sovol Official User Group: https://www.facebook.com/groups/sovol3d <https://www.facebook.com/sovol3d/ >
Facebook Page:https://www.facebook.com/sovol3d <https://twitter.com/Sovol3d >
Youtube: https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog <https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog >
------------------------------------------------------------------
From:ccatlett1984 ***@***.***>
Sent At:2023 May 23 (Tue.) 06:41
To:Sovol3d/SV06-PLUS ***@***.***>
Cc:info ***@***.***>; Comment ***@***.***>
Subject:Re: [Sovol3d/SV06-PLUS] Please enable Folder browsing support. (Issue #1)
Sorry for the typo, I mean the SD card.

**Comment by ccatlett1984** [comment]
No, it can NOT.
Folders do not appear at all in the file selection dialog screen.

**Comment by Sovol3d** [comment]
Hello,
May I know your order number?
Could you please take a short video to show it?
website: sovol3d.com
Follow us to get an exclusive 3D printing information
Sovol Official User Group: https://www.facebook.com/groups/sovol3d <https://www.facebook.com/sovol3d/ >
Facebook Page:https://www.facebook.com/sovol3d <https://twitter.com/Sovol3d >
Youtube: https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog <https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog >
------------------------------------------------------------------
From:ccatlett1984 ***@***.***>
Sent At:2023 May 27 (Sat.) 05:22
To:Sovol3d/SV06-PLUS ***@***.***>
Cc:info ***@***.***>; Comment ***@***.***>
Subject:Re: [Sovol3d/SV06-PLUS] Please enable Folder browsing support. (Issue #1)
No, it can NOT.
 Folders do not appear at all in the file selection dialog screen.

**Comment by TerkWerX** [comment]
ever fixed?

**Comment by ccatlett1984** [comment]
Nope, they don't seem to be working on it at all....

**Comment by smkent** [comment]
[In digging around the SV06 Plus firmware to get LCD progress working for USB prints](https://github.com/smkent/sovol-sv06-plus-firmware), I've learned a bit about how the LCD works and how file names are populated. The LCD is its own little microcontroller that talks to the printer's own firmware via a serial connection; it's not like a monitor driven by a display signal from a computer. What this means is the LCD's pages are all preprogrammed and the printer firmware tells the LCD which page to go to in response to certain events, with some fields for populating data (like file names).

The SV06 Plus LCD has 4 predefined file selection pages with 5 file name slots each, and the printer firmware correspondingly fills up to these [20 file name slots](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.h#L11) (with [up to 20 characters each](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.h#L10)) using [file names from the root directory of the SD card](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.cpp#L134-L157).

It might be possible to implement subdirectory navigation using the 4 predefined LCD file list screens. Navigation would need to be implemented in the firmware's `LCD_RTS.cpp` beyond the simple file listing logic that's already there.

**Comment by ccatlett1984** [comment]
Yep, saw that when i was converting my lcd/printer over to klipper.
